### PR TITLE
Fix openQA-client package for openSUSE

### PR DESCRIPTION
**Pull request by Martchus**
See
* https://progress.opensuse.org/issues/40898
* https://progress.opensuse.org/issues/38351

I executed `grep 'use ' $(rpm -ql openQA-client)` and added all modules we're using. I excluded those which are provided by an already existing `Requires`, `perl-base` or `perl`.

### CODE REVIEW

File: openQA.spec [hunk 1]
```diff
@@ -159,12 +159,16 @@ The openQA worker manages test engine (provided by os-autoinst package).
 %package client
 Summary:        Client tools for remote openQA management
 Group:          Development/Tools/Other
+Requires:       openQA-common = %{version}
```
Comment (coolo) [inline openQA.spec]: hmm, openQA-common is huge. Not a fan of this change

Comment (Martchus) [reply openQA.spec]: We could split openQA-common. But openQA-common requires only 830.3 KiB and its biggest dependency (Mojolicious) is required anyways. So I'd say it isn't worth the effort to save a few MB.

And I'd like to note that I don't introduce the dependency here. I just update the list of dependencies.

Comment (okurz) [reply openQA.spec]: no one said it's your fault ;)

Comment (coolo) [reply openQA.spec]: 'use openQA libs'? You're talking about 4745f7a38f45a1f72cb34397cd737453862e94cd, right? This means to me ./lib/OpenQA/UserAgent.pm is in the wrong package. It shouldn't be in common, but in client.

Comment (okurz) [reply openQA.spec]: Yes, that one

**Comment by Martchus** [comment]
I checked all `using`s and it seems the UserAgent is really only used by the client. So I did what @coolo suggested.

**Comment by okurz** [comment]
awesome! Maybe you want to try out extending https://github.com/os-autoinst/openQA/blob/master/openQA-test.spec to test the client package itself?

**Comment by okurz** [comment]
Yes, obviously you can not test the same within just one package. So I suggest to either parameterize the spec file for multiple pages or create spec-files per package or any other approach to proof installability.

**Comment by coolo** [comment]
@Martchus you failed to identify all modules, sorry :)

Can't locate OpenQA/Client/Upload.pm in @INC

**Comment by coolo** [comment]
And if I copy over the Client directory, I still get:

```
Can't locate OpenQA/File.pm in @INC (you may need to install the OpenQA::File module) (@INC contains: /usr/share/openqa/script/../lib /usr/lib/perl5/site_perl/5.26.2/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.26.2 /usr/lib/perl5/vendor_perl/5.26.2/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.26.2 /usr/lib/perl5/5.26.2/x86_64-linux-thread-multi /usr/lib/perl5/5.26.2 /usr/lib/perl5/site_perl/5.26.1/x86_64-linux-thread-multi /usr/lib/perl5/site_perl/5.26.1 /usr/lib/perl5/vendor_perl/5.26.1/x86_64-linux-thread-multi /usr/lib/perl5/vendor_perl/5.26.1 /usr/lib/perl5/site_perl) at /usr/share/openqa/script/../lib/OpenQA/Client/Upload.pm line 21.
BEGIN failed--compilation aborted at /usr/share/openqa/script/../lib/OpenQA/Client/Upload.pm line 21.
Compilation failed in require at /usr/share/openqa/script/../lib/OpenQA/Client.pm line 20.
BEGIN failed--compilation aborted at /usr/share/openqa/script/../lib/OpenQA/Client.pm line 20.
Compilation failed in require at /usr/bin/openqa-client line 143.
BEGIN failed--compilation aborted at /usr/bin/openqa-client line 143.
```

And there we're doomed.

**Comment by Martchus** [comment]
Maybe we should really follow this test package approach (creating one test package per split package).

We could move `OpenQA::Client::Upload` to the client package as well. But yes, the problem is that `OpenQA::File` is also used by `OpenQA::Worker::Jobs` and itself requires further modules like `OpenQA::Parser::Results` and everything what depends on it. So we can't simply move that one, too. So the dependency to the common package for the client stays.

**Comment by coolo** [comment]
sounds like it - moving the client specifics to -client sounds still reasonable. Testing the package in OBS is still not my preference, testing it in travis on PRs is the real goal.

**Comment by Martchus** [comment]
Ok, so something like this: https://github.com/os-autoinst/openQA/pull/1819 (not tested yet)

> testing it in travis on PRs is the real goal

Of course that would make sense. It would likely be possible to invoke `osc build` inside a docker container. Then we could install the resulting packages one after another to see if they work. Before installing the next package we could uninstall the previous with `zypper rm --clean-deps`.

**Comment by okurz** [comment]
I think https://github.com/openSUSE/openSUSE-release-tools/blob/master/dist/ci/obs-build-target is doing that in a nice way, testing the package and spec file but provide it with travis CI, right?

**Comment by coolo** [comment]
It's one way - if you check https://github.com/openSUSE/open-build-service/pull/6076 you'll see a OBS Package Build check, which is another

**Pull request merged**